src/app/games/diaballik/tests/DiaballikRules.spec.ts

```diff
@@ -293,4 +333,51 @@ describe('DiaballikRules', () => {
         // Then it should detect the victory
         RulesUtils.expectToBeVictoryFor(rules, node, Player.ONE);
     });
+    it('should detect non victories as ongoing', () => {
+        // Given a state without victory
+        const state: DiaballikState = DiaballikState.getInitialState();
+        const node: DiaballikNode = new DiaballikNode(state);
+        // When checking its status
+        // Then it should be ongoing
+        RulesUtils.expectToBeOngoing(rules, node);
+    });
+    describe('victory and blocker coords', () => {
+        it('should not detect anything on initial state', () => {
+            // Given a state without victory
+            const state: DiaballikState = DiaballikState.getInitialState();
+            // When computing victory/defeat coords
+            // Then it should have none
+            expect(DiaballikRules.get().getVictoryOrDefeatCoords(state).isAbsent()).toBeTrue();
+        });
+        it('should not detect blockers when there is a discontinuous line', () => {
```

Comment:
en titre clairs d'explication des règles je dirais
should not detect blockers when goal line is accessible
should not detect blockers when there is less than three blocked player
should detect blockers when there is three or more blocked players ?

 En fait içi c'est p-e plus describe 'getVictoryOrDefeatCoords' que les cas de victoires à tester non ?

Reply:
oui exactement c'est juste du test de getVictoryOrDefeatCoords (et uniquement pour la couverture). Les vrais tests de fonctionnement sont faits dans le composant, où on regarde que c'est vraiment highlighté. Mais s'il y a pas de victoire, getVictoryOrDefeatCoords n'est pas appelé, donc ici ce sont uniquement des tests de couverture.